apps/s_client.c

```diff
@@ -446,13 +446,11 @@ static int tlsa_import_rr(SSL *con, const char *rrdata)
     OPENSSL_free(data);
 
     if (ret == 0) {
-        ERR_print_errors(bio_err);
         BIO_printf(bio_err, "%s: warning: unusable TLSA rrdata: %s\n",
                    prog, rrdata);
         return 0;
     }
     if (ret < 0) {
-        ERR_print_errors(bio_err);
```

Comment:
These two deletions look suspect to me. The next lines indicate that problems here are considered warnings only and processing will continue - so we risk losing error data associated with the warning.

Reply:
Good catch - if the `rrset` has more than one element and things work fine for at least one of them but go wrong for at least one other element, that error detail would get lost when removing the `ERR_print_errors(bio_err)`.
Reverted the removal here and in the case just above it.